### PR TITLE
[CWS] Add `is_public` field to IP addresses

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

This PR allows evaluating whether an IP address is public using a list of pre-defined private IP ranges.

### CODE REVIEW

File: pkg/security/probe/field_handlers.go [hunk 1]
```diff
@@ -74,3 +76,54 @@ func getProcessService(config *config.Config, entry *model.ProcessCacheEntry) st
...
+	for _, cidr := range cfg.Probe.NetworkPrivateIPRanges {
+		if err := bfh.privateCIDRs.AppendIP(cidr); err != nil {
+			return nil, fmt.Errorf("error adding private IP range %s: %w", cidr, err)
+		}
+	}
+	for _, cidr := range cfg.Probe.NetworkExtraPrivateIPRanges {
+		if err := bfh.privateCIDRs.AppendIP(cidr); err != nil {
+			return nil, fmt.Errorf("error adding extra private IP range %s: %w", cidr, err)
+		}
+	}
```
Comment (spikat) [inline pkg/security/probe/field_handlers.go:103]: Shouldn't be `AppendCIDR` instead of `AppendIP` ?

**Review by spikat — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]

File: pkg/config/setup/system_probe.go [hunk 2]
```diff
@@ -450,3 +452,27 @@ func eventMonitorBindEnv(config pkgconfigmodel.Config, key string) {
...
+	"10.0.0.0/8",
+	"172.16.0.0/12",
+	"192.168.0.0/16",
+	// IETF RFC 5735
+	"0.0.0.0/8",
+	"127.0.0.0/8",
+	"169.254.0.0/16",
+	"192.0.0.0/24",
+	"192.0.2.0/24",
+	"192.88.99.0/24",
```
Comment (brycekahle) [inline pkg/config/setup/system_probe.go:468]: This block is not private:
> In contrast with previously described blocks, packets destined to addresses from this block do appear in the public Internet.

**Review by brycekahle — APPROVED** [review]
One minor comment

File: pkg/config/setup/system_probe.go (hunk 2 above)
Comment (YoannGh) [reply pkg/config/setup/system_probe.go]: Removed it, thanks!

**Pull request merged**